src/dmd/denum.d

```diff
@@ -65,6 +66,8 @@ extern (C++) final class EnumDeclaration : ScopeDsymbol
         //printf("EnumDeclaration() %s\n", toChars());
         type = new TypeEnum(this);
         this.memtype = memtype;
+        if (memtype)
+            hasUserDefinedType = true;
```

Comment:
This introduces a very subtle bug. Take a look at the function getMemType(). If memtype is not provided, it is set to a default value. Now take a look at syntaxCopy(), which creates a new EnumDeclaration and passes memtype to it. But if memtype was set by getMemType(), it is not null, and hasUserDefinedType is incorrectly set to true.

The easiest way to fix this is have syntaxCopy() also copy hasUserDefinedType.

Reply:
Or more properly, make hasUserDefinedType a parameter to the constructor.